src/iosource/BPF_Program.cc

```diff
@@ -120,12 +120,19 @@ bool BPF_Program::Compile(int snaplen, int linktype, const char* filter, uint32_
 	int err = pcap_compile_nopcap(snaplen, linktype, &m_program, (char*)filter, optimize, netmask,
 	                              my_error);
 	if ( err < 0 )
-		errbuf = std::string(my_error);
+		{
+		state = GetStateFromMessage(errstr);
```

Comment:
`errstr` isn't defined here. Should that be std::string(my_error) instead?

But then, my Debian box, a Centos-7 container and FreeBSD don't seems to have that error parameter and all CI is green as well, is this actually live code anywhere? Custom libpcap?

Maybe we can replace this code with`#error Please report to use what you're running` or reject configs with `LIBPCAP_PCAP_COMPILE_NOPCAP_HAS_ERROR_PARAMETER` further up rather than have it untested?

Reply:
You also got me wondering whether `pcap_compile_nopcap` sets `pcap_error`, and since it just calls `pcap_compile` underneath it should. That means we do actually get an error string back from it similar to the `pcap_compile` version of the method.

Reply:
> We'll find out fairly quickly if someone is using it. In the meantime we could leave the code in place #ifdef'd like it is, and make a note to remove it at a later date.

Resolving this thread as it's gone now. The `LIBPCAP_PCAP_COMPILE_NOPCAP_HAS_ERROR_PARAMETER` vestigial are still in cmake, but doesn't need to hold up this one.